### PR TITLE
fix: Throttle storageCache population

### PR DESCRIPTION
**Pull request by adityapk00**
## Motivation

When a hub starts up, it attempts to pre-populate the storage cache. This is slow, since it has to basically scan every message. 

## Change Summary

- Slow down the pre-population
- Don't do too many scans in parallel.

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [X] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [X] PR has a [changeset](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#35-adding-changesets)
- [X] PR has been tagged with a change label(s) (i.e. documentation, feature, bugfix, or chore)
- [ ] PR includes [documentation](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#32-writing-docs) if necessary.
- [X] All [commits have been signed](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#22-signing-commits)

## Additional Context

If this is a relatively large or complex change, provide more details here that will help reviewers

---

## PR-Codex overview
This PR focuses on fixing storage cache prepopulation throttling and adding memory usage logging with optional heap dumps. 

### Detailed summary
- Fixed throttle issue in storage cache prepopulation
- Added memory usage logging in LibP2PNode and Hub
- Implemented optional heap dumps for memory analysis

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/hubble/src/storage/stores/storageCache.ts [hunk 1]
```diff
@@ -93,7 +96,9 @@ export class StorageCache {
       log.error("cannot prepopulate message counts, db is not open");
       throw new HubError("unavailable.storage_failure", "cannot prepopulate message counts, db is not open");
     }
-    this.prepopulateMessageCounts();
+
+    // Run the prepopulation without waitinf for it to finish
```
Comment (Wazzymandias) [inline apps/hubble/src/storage/stores/storageCache.ts:100]:
```suggestion
    // Run the prepopulation without waiting for it to finish
```

File: apps/hubble/src/storage/stores/storageCache.ts [hunk 2]
```diff
@@ -143,6 +148,13 @@ export class StorageCache {
   async getMessageCount(fid: number, set: UserMessagePostfix, forceFetch = true): HubAsyncResult<number> {
     const key = makeKey(fid, set);
     if (this._counts.get(key) === undefined && forceFetch) {
+      if (this._pendingMessageCountScans > MAX_PENDING_MESSAGE_COUNT_SCANS) {
+        log.error({ pendingScans: this._pendingMessageCountScans }, "too many pending message count scans");
+        return err(new HubError("unavailable.storage_failure", "too many pending message count scans"));
```
Comment (Wazzymandias) [inline apps/hubble/src/storage/stores/storageCache.ts:153]: noob question: Would it be problematic if this returns error and add message ends up updating the map to set count to zero in `storageCache.ts:addMessage` ? 

```
      const count = this._counts.get(key) ?? (await this.getMessageCount(fid, set)).unwrapOr(0);
```

Comment (adityapk00) [reply apps/hubble/src/storage/stores/storageCache.ts:153]: Hmm, this is another bug. I'll change it to not update the cache on error in `addMessage`. Thanks!

File: apps/hubble/src/storage/stores/storageCache.ts [hunk 3]
```diff
@@ -260,7 +274,16 @@ export class StorageCache {
       const set = typeToSetPostfix(message.data.type);
       const fid = message.data.fid;
       const key = makeKey(fid, set);
-      const count = this._counts.get(key) ?? (await this.getMessageCount(fid, set)).unwrapOr(0);
+      let count = this._counts.get(key);
```
Comment (Wazzymandias) [inline apps/hubble/src/storage/stores/storageCache.ts:277]: Should we do similar in `removeMessage` ? 

```
  private async removeMessage(message: Message): Promise<void> {
    if (message.data !== undefined) {
      const set = typeToSetPostfix(message.data.type);
      const fid = message.data.fid;
      const key = makeKey(fid, set);
      const count = this._counts.get(key) ?? (await this.getMessageCount(fid, set)).unwrapOr(0);
      if (count === 0) {
        log.error(`error: ${set} store message count is already at 0 for fid ${fid}`);
      } else {
        this._counts.set(key, count - 1);
      }
```

Comment (adityapk00) [reply apps/hubble/src/storage/stores/storageCache.ts:277]: The bug is not present in removeMessage, but maybe we should do it for consistency?

**Review by Wazzymandias — APPROVED** [review]

**Pull request merged**